### PR TITLE
Add 'get_ann_info' to dataset_wrappers

### PR DESCRIPTION
**Pull request by zhaoxin111**
## Motivation

When the parameter dataset in MultiImageMixDataset is ConcatDataset/RepeatDataset/ClassBalancedDataset and mixup is added in the pipeline, the get_indexes in mixup will crash since there are no get_ann_info in ConcatDataset/RepeatDataset/ClassBalancedDataset.

I submitted a PR before, but it was closed by me. I added a unit test on the basis of the previous PR
Related PR：
https://github.com/open-mmlab/mmdetection/pull/6352
https://github.com/open-mmlab/mmdetection/pull/6516
## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RangiLyu** [comment]
Lint failed. Please install pre-commit and then run pre-commit run --all-files to fix.

**Comment by zhaoxin111** [comment]
> Lint failed. Please install pre-commit and then run pre-commit run --all-files to fix.

Hi, I tried pre_commit, but the pyflakes version is always wrong. I tried to install the 2.2.0 version, but this is still wrong. Do you have any suggestions?
image

File: mmdet/datasets/dataset_wrappers.py [hunk 1]
```diff
@@ -68,6 +68,30 @@ def get_cat_ids(self, idx):
...
+        if idx < 0:
+            if -idx > len(self):
+                raise ValueError(
+                    'absolute value of index should not exceed dataset length')
+            idx = len(self) + idx
+        dataset_idx = bisect.bisect_right(self.cumulative_sizes, idx)
+        if dataset_idx == 0:
+            sample_idx = idx
+        else:
+            sample_idx = idx - self.cumulative_sizes[dataset_idx - 1]
```
Comment (ZwwWayne) [inline mmdet/datasets/dataset_wrappers.py:92]: line 83-92 can be encapsulated into a function like get_sample_idx and we can use it in many places.

Comment (zhaoxin111) [reply mmdet/datasets/dataset_wrappers.py:92]: This function is only used in two places. Where should this function be placed?

File: mmdet/datasets/dataset_wrappers.py [hunk 2]
```diff
@@ -68,6 +68,30 @@ def get_cat_ids(self, idx):
             sample_idx = idx - self.cumulative_sizes[dataset_idx - 1]
         return self.datasets[dataset_idx].get_cat_ids(sample_idx)
 
+    def get_ann_info(self, idx):
+        """Get annotation of concatenated dataset by index.
+
+        This is needed by MixUp.
```
Comment (RangiLyu) [inline mmdet/datasets/dataset_wrappers.py:74]: This line is not necessary.

**Review by RangiLyu — APPROVED** [review]

**Comment by RangiLyu** [comment]
ping @hhaAndroid @ZwwWayne

**Review by hhaAndroid — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**